presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -149,99 +200,186 @@ public BaseJdbcClient(JdbcConnectorId connectorId, BaseJdbcConfig config, String
     }
 
     @Override
+    public Set<String> getSchemaNames()
+    {
+        ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+        Set<String> originalNames = getOriginalSchemas();
+        Map<String, Optional<String>> mappedNames = new HashMap<>();
+        for (String schemaName : originalNames) {
+            String schemaNameLower = schemaName.toLowerCase(ENGLISH);
+            schemaNames.add(schemaNameLower);
+            mappedNames.put(schemaNameLower, Optional.of(schemaName));
+        }
+
+        // if someone is listing all of the schema names, throw them all into the cache as a refresh since we already spent the time pulling them from the DB
+        schemaMappingCache.putAll(mappedNames);
+        return schemaNames.build();
+    }
+
+    /**
+     * Pull the list of Table names from the RDBMS exactly as they are returned. Override if the RDBMS method to pull the tables is different. Each {@link String} array must have
+     * the table's original schema name as the first element, and the original table name as the second element.
+     *
+     * @param schema the schema to list the tables for, or NULL for all
+     * @return the schema + table names
+     */
+    protected List<String[]> getOriginalTablesWithSchema(Connection connection, String schema)
```

Comment:
`SchemaTableName` in it's constructor specifically lowercases both the schema and the table strings, thus can't be used for this.

I chose an array as an easy, simple, basic structure that wouldn't take much effort. Also can't be a Map since a schema can have a multiple tables, and we might be pulling from multiple schemas at once rather than just 1 schema.